src/plugins/dashboard/public/services/dashboard_backup_service.ts

```diff
@@ -31,34 +28,39 @@ const DASHBOARD_VIEWMODE_LOCAL_KEY = 'dashboardViewMode';
 // this key is named `panels` for BWC reasons, but actually contains the entire dashboard state
 const DASHBOARD_STATE_SESSION_KEY = 'dashboardStateManagerPanels';
 
-interface DashboardBackupRequiredServices {
-  notifications: DashboardNotificationsService;
-  spaces: DashboardSpacesService;
+interface DashboardBackupServiceType {
+  clearState: (id?: string) => void;
+  getState: (id: string | undefined) =>
+    | {
+        dashboardState?: Partial<SavedDashboardInput>;
+        panels?: UnsavedPanelState;
+      }
+    | undefined;
+  setState: (
+    id: string | undefined,
+    dashboardState: Partial<SavedDashboardInput>,
+    panels: UnsavedPanelState
+  ) => void;
+  getViewMode: () => ViewMode;
+  storeViewMode: (viewMode: ViewMode) => void;
+  getDashboardIdsWithUnsavedChanges: () => string[];
+  dashboardHasUnsavedEdits: (id?: string) => boolean;
 }
 
-export type DashboardBackupServiceFactory = KibanaPluginServiceFactory<
-  DashboardBackupServiceType,
-  DashboardStartDependencies,
-  DashboardBackupRequiredServices
->;
-
 class DashboardBackupService implements DashboardBackupServiceType {
   private activeSpaceId: string;
   private sessionStorage: Storage;
   private localStorage: Storage;
-  private notifications: DashboardNotificationsService;
-  private spaces: DashboardSpacesService;
 
   private oldDashboardsWithUnsavedChanges: string[] = [];
 
-  constructor(requiredServices: DashboardBackupRequiredServices) {
-    ({ notifications: this.notifications, spaces: this.spaces } = requiredServices);
+  constructor() {
     this.sessionStorage = new Storage(sessionStorage);
     this.localStorage = new Storage(localStorage);
 
     this.activeSpaceId = 'default';
-    if (this.spaces.getActiveSpace$) {
-      firstValueFrom(this.spaces.getActiveSpace$()).then((space) => {
+    if (spacesService?.getActiveSpace$) {
```

Comment:
Should this just check for spacesService?

Reply:
Good call :+1: Done in 299a41909872436fdf6d0fbcbcf139490cc70b3d